tests/test_init.py

```diff
@@ -53,3 +60,102 @@ def test_parse_bad_patchwork_url(self):
             "Can't parse patchwork url: '{}'".format(patchwork_url)
             in context.exception
         )
+
+
+class KBuilderTest(unittest.TestCase):  # pylint: disable=R0902
+    """Test cases for skt.kbuilder class"""
+    def setUp(self):
+        self.tmpdir = tempfile.mkdtemp()
+        self.tmpconfig = tempfile.NamedTemporaryFile()
+        self.kbuilder = skt.kbuilder(self.tmpdir, self.tmpconfig.name)
+        self.m_popen = Mock()
+        self.m_popen.returncode = 0
+        self.ctx_popen = mock.patch('subprocess.Popen',
+                                    Mock(return_value=self.m_popen))
+        self.ctx_check_call = mock.patch('subprocess.check_call', Mock())
+        self.m_io_open = Mock()
+        self.m_io_open.__enter__ = lambda *args: self.m_io_open
+        self.m_io_open.__exit__ = lambda *args: None
+        self.ctx_io_open = mock.patch('io.open',
+                                      Mock(return_value=self.m_io_open))
+        self.kernel_tarball = 'linux-4.16.0.tar.gz'
+        self.success_str = 'Tarball successfully created in ./{}\n'
+        self.success_str = self.success_str.format(self.kernel_tarball)
+
+    def tearDown(self):
+        shutil.rmtree(self.tmpdir)
+
+    def test_mktgz_clean(self):
+        """Check if `mrproper` target is called when clean is True"""
+        with self.ctx_popen, self.ctx_check_call as m_check_call:
+            self.assertRaises(Exception, self.kbuilder.mktgz, clean=True)
+            self.assertEqual(
+                m_check_call.mock_calls[0],
+                mock.call(['make', '-C', self.tmpdir, 'mrproper'])
+            )
+            m_check_call.reset_mock()
+            self.assertRaises(Exception, self.kbuilder.mktgz, clean=False)
+            self.assertEqual(m_check_call.call_count, 2)
+            self.assertNotEqual(
+                m_check_call.mock_calls[0],
+                mock.call(['make', '-C', self.tmpdir, 'mrproper']),
+            )
```

Comment:
This still tests the exact arguments, which is making the test fragile. Do you think you can test that argv[0] == "make" and argv[something above zero] == "mrproper", and the reverse?